community/modules/compute/schedmd-slurm-gcp-v5-node-group/variables.tf

```diff
@@ -73,37 +73,35 @@ variable "metadata" {
   default     = {}
 }
 
-variable "source_image_project" {
-  type        = string
+variable "instance_image" {
   description = <<-EOD
-    Project path where the source image comes from. If not provided, this value
-    will default to the project hosting the slurm-gcp public images. More
-    information can be found in the slurm-gcp docs:
+    Defines the image that will be used in the node group VM instances. If not
+    provided, the Slurm on GCP default published images will be used.
+
+    Expected Fields:
+    name: The name of the image. Mutually exclusive with family.
+    family: The image family to use. Mutually exclusive with name.
+    project: The project where the image is hosted.
+
+    Custom images must comply with Slurm on GCP requirements; it is highly
+    advised to use the packer templates provided by Slurm on GCP when
+    constructing custom slurm images.
+    
+    More information can be found in the slurm-gcp docs:
     https://github.com/SchedMD/slurm-gcp/blob/v5.1.0/docs/images.md#public-image.
     EOD
-  default     = null
-}
-
-variable "source_image_family" {
-  type        = string
-  description = <<-EOD
-    Source image family. If not provided, the default image family name for the
-    hpc-centos-7 version of the slurm-gcp public images will be used. More
-    information can be found in the slurm-gcp docs:
-    https://github.com/SchedMD/slurm-gcp/blob/v5.1.0/docs/images.md#public-image
-    EOD
-  default     = null
-}
+  type        = map(string)
+  default     = {}
 
-variable "source_image" {
-  type        = string
-  description = <<-EOD
-    Source disk image. By default, the image used will be the hpc-centos7
-    version of the slurm-gcp public images. More information can be found in the
-    slurm-gcp docs:
-    https://github.com/SchedMD/slurm-gcp/blob/v5.1.0/docs/images.md#public-image
-    EOD
-  default     = null
+  validation {
+    condition = length(var.instance_image) == 0 || (
+    can(var.instance_image["family"]) || can(var.instance_image["name"])) == can(var.instance_image["project"])
+    error_message = "The project is required if family or name are provided in var.instance_image."
+  }
+  validation {
+    condition     = length(var.instance_image) == 0 || can(var.instance_image["family"]) != can(var.instance_image["name"])
+    error_message = "Only one of family and name can be provided in var.instance_image."
```

Comment:
The problem here is that if I pass `{ dog: good }` into here, validation passes. 

Maybe just single validation:
```
length(var.instance_image) == 0 ||
(can(family) && can(project)) ||
(can(name)&& can(project))
```

Reply:
It was caught for me:
```
$ terraform -chdir=im-test/primary apply
╷
│ Error: Invalid value for variable
│ 
│   on main.tf line 96, in module "node_group4":
│   96:   instance_image = {
│   97:     dog = "good"
│   98:   }
│ 
│ Only one of family and name can be provided in var.instance_image.
│ 
│ This was checked by the validation rule at
│ modules/schedmd-slurm-gcp-v5-node-group/variables.tf:101,3-13.
```

The error message could stand to be updated though. If both name and family are not set the second validation will fail.

Reply:
Updated the error messages in the latest commit.

Reply:
You are correct. I did not actually run the code. Looks good.